homeassistant/components/mqtt/sensor.py

```diff
@@ -239,6 +239,12 @@ def _update_state(msg: ReceiveMessage) -> None:
                 self._attr_native_value = new_value
                 return
             try:
+                if (dt_util.parse_datetime(new_value)) is None and (
```

Comment:
The same check is also at L248.

Reply:
In this line (L242) I need to check if the value cannot be converted with `parse_datetime` to apply the conversion with `fromtimestamp` and provide a new converted variable `new_value`.

On line 248 I convert the variable `new_value` (from the payload or from the conversion of `fromtimestamp`).

So in each of these lines I analyze different things.

Do you have a suggestion to improve this?

Reply:
Followed the suggestion made below with https://github.com/home-assistant/core/pull/109773/commits/f40d99f49a37bafbdda9877acb9dfb0130603c95